include/rocksdb/utilities/transaction_db.h

```diff
@@ -240,6 +241,14 @@ struct TransactionDBOptions {
   // user-defined timestamps so this option only applies in this case.
   bool enable_udt_validation = true;
 
+  //       / \     UNDER CONSTRUCTION
+  //      / ! \    UNDER CONSTRUCTION
+  //     /-----\   UNDER CONSTRUCTION
+  //
+  // The secondary indices to be maintained. See the SecondaryIndex interface
+  // for more details.
+  std::vector<std::shared_ptr<SecondaryIndex>> secondary_indices;
```

Comment:
Maybe we can add a note that once this field is configured, entries that are not considered indexable by any of these secondary indices cannot be added anymore.

Reply:
Hm, not sure I follow... An index not being applicable to a certain primary-key value (e.g. because it's not written to the index's primary CF, or it doesn't have the indexing column) should not be an error (unless I have a bug somewhere). For example, "key1" in the test is initially not indexable (it doesn't have the right column) but can still be successfully written.

Reply:
Yeah, I just mean that if any of the applied `SecondaryIndex` APIs return an non-OK status, this entry will not be committed. Something like what this test code does:

```
 if (primary_column_value == "corge") {
        return Status::NotSupported();
      }
```

Maybe "not indexable" is not a good description, but something along those lines.

Reply:
Oh you mean when an index is applicable but the indexing fails. That's already documented in the `SecondaryIndex` API comments I think ("Returning a non-OK status...")